src/storage/v2/transaction.hpp

```diff
@@ -31,12 +31,14 @@ const uint64_t kTimestampInitialId = 0;
 const uint64_t kTransactionInitialId = 1ULL << 63U;
 
 struct Transaction {
-  Transaction(uint64_t transaction_id, uint64_t start_timestamp, IsolationLevel isolation_level)
+  Transaction(uint64_t transaction_id, uint64_t start_timestamp, IsolationLevel isolation_level,
+              StorageMode storage_mode)
```

Comment:
I understand why you need StorageMode in Transaction because of deltas but to me at least, it doesn't look like a perfect solution that the transaction knows about the StorageMode in which it operates. But I think Benjo is the perfect person to him about this so pls. double check :smile:

Reply:
@as51340 I think  I was wrong. It makes sense to not store in transactions storage mode, as it is only a "global" state. I can refactor this because I see what you are saying and it should be good.

Reply:
Great, I think it would be best to hold storage mode just in one place because of simplicity, that's the main reason so if you can refactor this and you think it makes sense :+1:

Reply:
I think I was wrong, I left comment above, let me know what do you think.